.github/scripts/build_assets/SvgIdUpdater.py

```diff
@@ -0,0 +1,91 @@
+import re
+from pathlib import Path
+
+
+class SvgIdUpdater:
+    """
+    Update an svg's id reference so it's unique among other
+    svg files in this repo.
+    """
+
+    # look for anything that is between ` id="` and `"`
+    id_declare_pattern = r"(?<= id=(\"|')).+?(?=(\"|'))"
+
+    # look for anything that is between `"url(#` and `)"`
+    id_use_pattern = r"(?<=\"url\(#).+?(?=\)\")"
+
+    # pattern made up of both
+    id_pattern = f"{id_declare_pattern}|{id_use_pattern}"
+
+    def __init__(self):
+        """
+        The id generator that we will use throughout the process.
+        """
+        self.id_generator = None
+
+        """
+        Tracks the ids that we are swapping out.
+        """
+        self.old_to_new_ids_dict = None
+
+
+    def update_id(self, filepath: Path):
+        """
+        Update the ids inside an svg file.
+        """
+        self.reset(filepath.stem)
+
+        # can switch to other patterns if needed
+        search_regexp = re.compile(SvgIdUpdater.id_pattern)
+            
+        with open(filepath, "r+") as file:
+            print("Opening " + file.name)
+
+            new_content = search_regexp.sub(self.sub_id, file.read())
+
+            # write to a new file
+            # new_path = filepath.with_name(filepath.stem + "-written" + filepath.suffix)
+            # with open(new_path, "w") as new_file:
+            #     new_file.write(new_content)
+            #     print("Finished substituting. Writing to " + new_file.name)
+                
+            # overwrite same file
+            file.write(new_content)  
```

Comment:
When testing out the script, it doesn't overwrite the svg file but just appends to it. 
```suggestion
            file.seek(0) 
            file.write(new_content)
            file.truncate()
```

Reply:
Thanks for pointing this out. I also found this error yesterday when testing and forgot to push the change. The most recent commit should fix it (same code as yours)